query/query_impl.go

```diff
@@ -137,6 +124,7 @@ func NaiveQueryApply(q Query, qr Results) Results {
 		qr = NaiveOffset(qr, q.Offset)
 	}
 	if q.Limit != 0 {
+		// TODO: Offset?
```

Comment:
Yeah, that's broken... You can tell how often we use these...

Reply:
Fixed and added tests.